libraries/smtp_library.py

```
@@ -174,17 +174,38 @@ def open_smtp_connection(host, port):
-        if str(expected_code) in data:
-            break
+        try:
+            chunk = sock.recv(1024)
+            if not chunk:
+                break
+            data += chunk.decode('utf-8') if isinstance(chunk, bytes) else chunk
+            if str(expected_code) in data:
+                break
+        except socket.timeout:
+            logger.error("Timeout: no response with code %s" % expected_code)
+            raise Exception("Timeout waiting for SMTP response %s" % expected_code)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)